tests/Kernel/CorporateBlocks/CorporateBlocksFooterTest.php

```diff
@@ -5,316 +5,264 @@
-    // Override config "oe_corporate_blocks.ec_data.footer" with test.
-    /* @var $config_obj \Drupal\Core\Config\Config */
-    $config_obj = \Drupal::service('config.factory')->getEditable('oe_corporate_blocks.ec_data.footer');
-    $test_data = $this->getFixtureContent('ec_footer.yml');
-    $config_obj->setData($test_data);
-    $config_obj->save();
-
-    // Setup and render footer block.
-    $config = [
-      'id' => 'oe_corporate_blocks_ec_footer',
-      'label' => 'OpenEuropa footer block',
-      'provider' => 'oe_corporate_blocks',
-      'label_display' => '0',
-    ];
-
-    $render = $this->buildBlock('oe_corporate_blocks_ec_footer', $config);
-
-    $html = (string) $this->container->get('renderer')->renderRoot($render);
+  public function testEcFooterCoreBlockRendering(): void {
+    $test_data = [];
+    $html = $this->renderCorporateBlocksFooter('ec', $test_data);
     $crawler = new Crawler($html);
 
     // Make sure that footer block is present.
-    $actual = $crawler->filter('footer.ecl-footer');
+    $actual = $crawler->filter('footer.ecl-footer-core');
     $this->assertCount(1, $actual);
 
     // Make sure that footer block rendered correctly.
-    $actual = $crawler->filter('footer.ecl-footer div.ecl-footer__sections div.ecl-row section.ecl-footer__section');
-    $this->assertCount(3, $actual);
+    $actual = $crawler->filter('footer.ecl-footer-core .ecl-footer-core__container .ecl-footer-core__section');
+    $this->assertCount(4, $actual);
 
-    $first_column = $crawler->filter('footer.ecl-footer div.ecl-footer__sections div.ecl-row section.ecl-footer__section:nth-child(1)');
+    $section = $crawler->filter('footer.ecl-footer-core section.ecl-footer-core__section1');
 
-    $actual = $first_column->filter('h1.ecl-footer__section-title');
-    $this->assertEquals($test_data['about_ec_title'], trim($actual->text()));
+    $actual = $section->filter('a.ecl-footer-core__title');
+    $this->assertEquals($test_data['site_name']['label'], $actual->text());
+    $this->assertEquals($test_data['site_name']['href'], $actual->attr('href'));
+    $actual = $section->filter('div.ecl-footer-core__description');
+    $expected = new FormattableMarkup('This site is managed by the @name', ['@name' => 'ACP–EU Joint Assembly']);
+    $this->assertEquals($expected, $actual->text());
 
-    $actual = $first_column->filter('ul li:nth-child(1) > a');
-    $this->assertEquals($test_data['about_ec_links'][0]['href'], $actual->attr('href'));
-    $this->assertEquals($test_data['about_ec_links'][0]['label'], $actual->text());
+    $section = $crawler->filter('footer.ecl-footer-core section.ecl-footer-core__section2');
 
-    $actual = $first_column->filter('ul li:nth-child(2) > a');
-    $this->assertEquals($test_data['about_ec_links'][1]['href'], $actual->attr('href'));
-    $this->assertEquals($test_data['about_ec_links'][1]['label'], $actual->text());
+    $actual = $section->filter('ul li:nth-child(1) > a');
```

Comment:
All link lists are tested the same way but we repeat the same code over and over again, we could use iterations or helper methods to make it less cluttered.